assets/src/edit-story/components/panels/layer/useElements.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app';
+
+import IconBackground from './layer_bg.svg';
+import IconMedia from './layer_media.svg';
+import IconVideo from './layer_video.svg';
+import IconText from './layer_text.svg';
+
+function getIconForElementType( type ) {
+	switch ( type ) {
+		case 'video':
+			return <IconVideo />;
+		case 'text':
+			return <IconText />;
+		case 'image':
+		case 'shape':
+		case 'square': // this is only while square is a "shape"
+			return <IconMedia />;
+		default:
+			return <IconBackground />;
+	}
+}
+
+function useElements() {
+	const {
+		state: { currentPage, selectedElementIds },
+	} = useStory();
+
+	if ( ! currentPage ) {
+		return [];
+	}
+
+	let backgroundElement, otherElements;
+	const hasBackground = Boolean( currentPage.backgroundElementId );
+
+	if ( hasBackground ) {
+		[ backgroundElement, otherElements ] = currentPage.elements;
+	} else {
+		otherElements = currentPage.elements;
+		backgroundElement = { id: '' };
```

Comment:
A bit iffy on this. Can we avoid a surrogate element? Above I suggested maybe we should have a `type` for a layer object that could include a `element=null` or an existing element when defined?

Reply:
Great ideas. I had to spend quite some time thinking what you meant and how to do this properly.

I'll completely rewrite this to use a new "background" element, that's only used for the layer panel, which will in turn refer to either no element or another element. And move all the content and icon logic to the elements as suggested.

Reply:
Finally figured it out in 07f0e2a.